xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -450,9 +453,9 @@ namespace PVR
       if (bOnOff && !channel->IsRecording())
       {
         CEpgInfoTagPtr epgTag;
-        int iDuration = CServiceBroker::GetSettings().GetInt(CSettings::SETTING_PVRRECORD_INSTANTRECORDTIME);
+        int iDuration = m_settings.GetIntValue(CSettings::SETTING_PVRRECORD_INSTANTRECORDTIME);
```

Comment:
Why do you introduce this and all the stuff in the PVRSettings. Why do we need a container for the PVR related settings? If you want to get the current setting value, all you need to do this calling this or the other type related setting methods. What do you want to achieve with it?